src/impl/mozXhr.js

```diff
@@ -34,5 +34,10 @@ export default function mozXhrRequest(options) {
   }
   xhr.addEventListener('progress', onProgressEvent);
   xhr.addEventListener('loadend', onLoadEvent);
+  xhr.addEventListener('error', err => options.onComplete({
```

Comment:
please can you pull this out into a helper methods (like `onProgressEvent` and `onLoadEvent`) for consistency.